# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptLoader.cpp
# Commit: f0d873f64979
# Full Hash: f0d873f64979928a059b519ea8409dd3c5286d39
# Author: Yoshi Cheng-Hao Huang <allstars.chh@gmail.com>
# Date: 2025-10-03 20:53 +0000
# Regressor Bug: 1968890
# File Overlap Count: 5
# Overlapping Files: js/loader/ModuleLoadRequest.cpp, js/loader/ModuleLoaderBase.cpp, js/loader/ModuleLoadRequest.h, js/loader/ModuleLoaderBase.h, dom/script/ScriptLoader.cpp
# Description:
#   Bug 1968890 - Part 7: Add Cancel method. r=jonco
# ==============================================================================

diff -r f06186536887 -r f0d873f64979 dom/script/ScriptLoader.cpp
--- a/dom/script/ScriptLoader.cpp	Fri Oct 03 20:53:29 2025 +0000
+++ b/dom/script/ScriptLoader.cpp	Fri Oct 03 20:53:29 2025 +0000
@@ -4033,6 +4033,10 @@
   NS_ASSERTION(aRequest, "null request in stream complete handler");
   NS_ENSURE_TRUE(aRequest, NS_ERROR_FAILURE);
 
+  if (aRequest->IsCanceled()) {
+    return NS_BINDING_ABORTED;
+  }
+
   nsresult rv = VerifySRI(aRequest, aLoader, aSRIStatus, aSRIDataVerifier);
 
   if (NS_SUCCEEDED(rv)) {
@@ -4481,9 +4485,6 @@
     return aStatus;
   }
 
-  if (aRequest->IsCanceled()) {
-    return NS_BINDING_ABORTED;
-  }
   MOZ_ASSERT(aRequest->IsFetching());
   CollectScriptTelemetry(aRequest);
 